ipam/mas_test.go

```diff
@@ -0,0 +1,140 @@
+package ipam
+
+import (
+	"net"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func TestNewMasSource(t *testing.T) {
+	options := make(map[string]interface{})
+	mas, _ := newMasSource(options)
+
+	if runtime.GOOS == "windows" {
+		if mas.filePath != defaultWindowsFilePath {
+			t.Errorf("default file path set incorrectly")
+		}
+	} else {
+		if mas.filePath != defaultLinuxFilePath {
+			t.Errorf("default file path set incorrectly")
+		}
+	}
+	if mas.name != "MAS" {
+		t.Errorf("mas source Name incorrect")
+	}
+}
+
+func TestGetSDNInterfaces(t *testing.T) {
+	const validFileName = "testfiles/masInterfaceConfig.json"
+	const invalidFileName = "mas_test.go"
+	const nonexistentFileName = "bad"
+
+	interfaces, err := getSDNInterfaces(validFileName)
+	if err != nil {
+		t.Fatalf("failed to get sdn Interfaces from file: %v", err)
+	}
+
+	correctInterfaces := &NetworkInterfaces{
+		Interfaces: []Interface{
+			{
+				MacAddress: "000D3A6E1825",
+				IsPrimary:  true,
+				IPSubnets: []IPSubent{
+					{
+						Prefix: "1.0.0.0/12",
+						IPAddresses: []IPAddress{
+							{Address: "1.0.0.4", IsPrimary: true},
+							{Address: "1.0.0.5", IsPrimary: false},
+							{Address: "1.0.0.6", IsPrimary: false},
+							{Address: "1.0.0.7", IsPrimary: false},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if !reflect.DeepEqual(interfaces, correctInterfaces) {
+		t.Fatalf("Interface list did not match expected list. expected: %v, actual: %v", interfaces, correctInterfaces)
+	}
+
+	interfaces, err = getSDNInterfaces(invalidFileName)
+	if interfaces != nil || err == nil {
+		t.Fatal("didn't throw error on invalid file")
+	}
+
+	interfaces, err = getSDNInterfaces(nonexistentFileName)
+	if interfaces != nil || err == nil {
+		t.Fatal("didn't throw error on nonexistent file")
+	}
+}
+
+func TestPopulateAddressSpace(t *testing.T) {
+
+	hardwareAddress, _ := net.ParseMAC("00:0d:3a:6e:18:25")
+	localInterfaces := []net.Interface{{HardwareAddr: hardwareAddress, Name: "eth0"}}
+
+	local := &addressSpace{
+		Id:    LocalDefaultAddressSpaceId,
+		Scope: LocalScope,
+		Pools: make(map[string]*addressPool),
+	}
+
+	sdnInterfaces := &NetworkInterfaces{
+		Interfaces: []Interface{
+			{
+				MacAddress: "000D3A6E1825",
+				IsPrimary:  true,
+				IPSubnets: []IPSubent{
+					{
+						Prefix: "1.0.0.0/12",
+						IPAddresses: []IPAddress{
+							{Address: "1.1.1.5", IsPrimary: true},
+							{Address: "1.1.1.6", IsPrimary: false},
+							{Address: "1.1.1.6", IsPrimary: false},
```

Comment:
can we have one more valid address and check the length of address records as 2?